lost/src/main/java/com/mapzen/android/lost/internal/LocationSettingsResultRequest.java

```diff
@@ -0,0 +1,218 @@
+
+  @NonNull @Override public LocationSettingsResult await() {
+    return generateLocationSettingsResult();
+  }
+
+  @NonNull @Override public LocationSettingsResult await(long time, @NonNull TimeUnit timeUnit) {
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    Future<LocationSettingsResult> future = executor.submit(new Callable() {
+      public LocationSettingsResult call() throws Exception {
+        return generateLocationSettingsResult();
+      }
+    });
+    LocationSettingsResult result;
+    try {
+      result = future.get(time, timeUnit);
+    } catch (TimeoutException e) {
+      result = createResultForStatus(TIMEOUT);
+    } catch (InterruptedException e) {
+      result = createResultForStatus(INTERRUPTED);
+    } catch (ExecutionException e) {
+      result = createResultForStatus(INTERNAL_ERROR);
+    }
+    executor.shutdownNow();
+
+    return result;
+  }
+
+  private LocationSettingsResult createResultForStatus(int statusType) {
+    Status status = new Status(statusType);
+    return new LocationSettingsResult(status, null);
+  }
+
+  @Override public void cancel() {
+    if (locationResultTask != null && !locationResultTask.isCancelled()) {
+      locationResultTask.cancel(true);
+    }
+  }
+
+
+  @Override public boolean isCanceled() {
+    return locationResultTask.isCancelled();
+  }
+
+  @Override
+  public void setResultCallback(@NonNull ResultCallback<? super LocationSettingsResult> callback) {
+    resultCallback = callback;
+    locationResultTask.execute();
+  }
+
+  @Override
+  public void setResultCallback(@NonNull ResultCallback<? super LocationSettingsResult> callback,
+      long time, @NonNull TimeUnit timeUnit) {
+    resultCallback = callback;
+    locationResultTask.execute(time, timeUnit);
+  }
+
+  private LocationSettingsResult generateLocationSettingsResult() {
+    boolean needGps = false;
+    boolean needNetwork = false;
+    for (LocationRequest request : settingsRequest.getLocationRequests()) {
+      switch (request.getPriority()) {
+        case LocationRequest.PRIORITY_HIGH_ACCURACY:
+          needGps = true;
+          needNetwork = true;
+          break;
+        case LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY:
+        case LocationRequest.PRIORITY_LOW_POWER:
+          needNetwork = true;
+          break;
+        default:
+          break;
+      }
+    }
+    boolean needBle = settingsRequest.getNeedBle();
+
+
+    boolean gpsUsable = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+    boolean gpsPresent = packageManager.hasSystemFeature(PackageManager.FEATURE_LOCATION_GPS);
+    boolean networkUsable = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+    boolean networkPresent = packageManager.hasSystemFeature(
+        PackageManager.FEATURE_LOCATION_NETWORK);
+    boolean bleUsable = bluetoothAdapter.isEnabled();
+    boolean blePresent = packageManager.hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE);
```

Comment:
`PackageManager.FEATURE_BLUETOOTH_LE` requires API 18. The current minimum SDK for LOST is 15.